db/migrate/20170413020922_add_uuid_fields_to_tables.rb

```diff
@@ -0,0 +1,14 @@
+class AddUuidFieldsToTables < ActiveRecord::Migration
+  TABLES_FOR_MIGRATION = [:answers, :assignments, :broadcast_addressings, :broadcasts, :choices, :conditions,
+    :form_forwardings, :form_items, :form_versions, :forms, :media_objects, :missions, :operations,
+    :option_nodes, :option_sets, :options, :questions, :report_calculations, :report_option_set_choices,
+    :report_reports, :responses, :settings, :sms_messages, :taggings, :tags,
+    :user_group_assignments, :user_groups, :users, :whitelistings]
+
+  def change
+    TABLES_FOR_MIGRATION.each do |table_name|
+      add_column table_name, :uuid, :string, null: false
```

Comment:
I did test this on existing data, though not enough of it to catch the other problem you mentioned with sphinx.